digits/model/forms.py

```diff
@@ -81,6 +81,17 @@ def validate_NetParameter(form, field):
                 ],
             )
 
+    ### Solver types
+
+    solver_type = wtforms.SelectField('Solver type',
+        choices = [
+                ('SGD', 'Stochastic gradient descent'),
+                ('ADAGRAD', 'Adaptive gradient'),
+                ('NESTEROV', 'Nesterov`s accelerated gradient'),
+                ],
+            default = 'SGD'
+            )
```

Comment:
How about this for the names:

    ('SGD', 'Stochastic gradient descent (SGD)'),
    ('ADAGRAD', 'Adaptive gradient (AdaGrad)'),
    ('NESTEROV', "Nesterov's accelerated gradient"),